truss/templates/control/control/helpers/inference_server_controller.py

```diff
@@ -142,13 +146,15 @@ def _patch_sort_key_fn(patch: Patch) -> int:
     # model code patches, so it's better to apply model code patches last to
     # avoid ending up with partially applied patches which currently triggers
     # full deploy.
-    if patch.type == PatchType.SYSTEM_PACKAGE:
-        return 0
-
-    if patch.type == PatchType.PYTHON_REQUIREMENT:
-        return 1
-
-    if patch.type == PatchType.MODEL_CODE:
-        return 2
 
-    raise ValueError(f"Unexpected patch type {patch.type}")
+    PATCH_ORDER = [
```

Comment:
great! thanks for adding this.

putting it like this is making me think: my gut feeling was that ENV_VARS would be before SYS_PACKAGES because the system packages install might rely on ENV_VARS (edge case I guess). But the way we're patching env vars (which is the way we should patch them) goes only to the control server.

Makes me think if we should run the python and system installs int he same os env as the iference server. 

I think not, though, because in the non-control server case the system and python packges happen in the docker build, which is agnostic? to the env. Actually it depends on where we set the env vars relative to the install scripts. Lemme check that

https://github.com/basetenlabs/truss/blob/main/truss/templates/base.Dockerfile.jinja#L62

Okay great. So the env vars are ADDED after the requirements and sys_packages are added and this is consistent.

EXTERNAL_DATA (like the model code and config), on the other hand, runs in the app_copy block of the docker file, which is actually before ENV_VARS.

So after this whole comment, I can finally suggest that we move the ENV_VARS to the end (after the config, model code, and external data) to be consistent with the dockerfile as it stands today, then we can revisit that question I opened with later on if we should consider them when running installs at a later time.

Thanks for coming to my TED Talk.

Reply:
sounds good – i'll update the ordering for consistency with original builds